onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc

```diff
@@ -1038,16 +1224,43 @@ static void TransposeInputImpl(api::GraphRef& graph, api::NodeRef& node, size_t
     // Permute1DConstant permutes the constant and adds a new initializer. The old initializer is removed only if
     // there are no other consumers.
     if (constant->Shape().size() == 1 && constant->Shape()[0] == gsl::narrow_cast<int64_t>(perm.size())) {
-      auto& node_to_update = dq_node ? *dq_node : node;
-      Permute1DConstant(graph, node_to_update, *constant, i, constant_to_modify, perm);
+      // TODO(adrianlizarraga): Remove handling of a DQ node from this special case.
+      // A quantized (roi/scales/sizes) input for Resize or a quantized pads input for Pad would be unlikely.
+      // HandleResize/HandlePad permute these kinds of inputs directly and do not try to call TransposeInput on them.
+      // So, we should not have a DQ to look past at this point, but we can handle it until verified.
```

Comment:
Should be able to check code coverage from existing unit tests. If nothing hits the code for a dq_node being present we can probably assume it's not required. 

If using VS \onnxruntime\VSCodeCoverage.runsettings can be used. Run the unit tests and it should give nice highlighting of the lines that are/aren't hit.

Reply:
Thanks! I checked code coverage from existing unit tests, and we never enter this special case with a DQ to look past. So, I replaced this logic with an `assert(!dq_to_look_past)`. I left a comment explaining why the callers to `TransposeInput` will not create this scenario.